### PR TITLE
feat: less to component token for List

### PR DESCRIPTION
**Pull request by jrr997**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
ref https://github.com/ant-design/ant-design/issues/41884

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | less to component token for List |
| 🇨🇳 Chinese | List组件less变量迁移 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 50570bf</samp>

Refactored list component style tokens and updated migration guide for less variables. The changes allow users to customize the list component theme and improve its responsiveness. The migration guide documents the new tokens and the deprecated less variables in both English and Chinese.

### <samp>🤖 Generated by Copilot at 50570bf</samp>

*  Add `listItemPadding`, `listItemPaddingSM`, and `listItemPaddingLG` properties to `ComponentToken` interface to enable list item padding customization for different sizes ([link](https://github.com/ant-design/ant-design/pull/42041/files?diff=unified&w=0#diff-8ed8eddb13b4537df4b99afb831a6d4be1bef5a7d3b3cc4ddac6b3d1307e87a4R8-R10))
* Remove `listItemPadding`, `listItemPaddingSM`, and `listItemPaddingLG` properties from `ListToken` interface as they are moved to `ComponentToken` interface ([link](https://github.com/ant-design/ant-design/pull/42041/files?diff=unified&w=0#diff-8ed8eddb13b4537df4b99afb831a6d4be1bef5a7d3b3cc4ddac6b3d1307e87a4L13-L15))
* Move `listItemPadding`, `listItemPaddingSM`, and `listItemPaddingLG` properties from `genListStyle` function to `withStyle` function to make them responsive to theme context ([link](https://github.com/ant-design/ant-design/pull/42041/files?diff=unified&w=0#diff-8ed8eddb13b4537df4b99afb831a6d4be1bef5a7d3b3cc4ddac6b3d1307e87a4L354-R363))
* Update English and Chinese migration guides for less variables to reflect changes in list component tokens and deprecation of some less variables ([link](https://github.com/ant-design/ant-design/pull/42041/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aR10-R26), [link](https://github.com/ant-design/ant-design/pull/42041/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739R10-R26))

### CODE REVIEW

File: docs/react/migrate-less-variables.zh-CN.md [hunk 1]
```diff
@@ -7,6 +7,23 @@ title: less 变量迁移 Component Token
...
+| `@list-footer-background` | - | 由于样式变化已废弃 |
+| `@list-empty-text-padding` | - | 由于样式变化已废弃 |
+| `@list-item-padding` | `listItemPadding` | - |
+| `@list-item-padding-sm` | `listItemPaddingSM` | - |
+| `@list-item-padding-lg` | `listItemPaddingLG` | - |
+| `@list-item-meta-margin-bottom` | - | 由于样式变化已废弃 |
+| `@list-item-meta-avatar-margin-right` | - | 由于样式变化已废弃 |
+| `@list-item-meta-title-margin-bottom` | - | 由于样式变化已废弃 |
+| `@list-customize-card-bg` | - | 由于样式变化已废弃 |
+| `@list-item-meta-description-font-size` | - | 由于样式变化已废弃 |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:25]:
```suggestion
| `@list-header-background` | - | 由于样式变化已废弃 |
| `@list-footer-background` | - | 由于样式变化已废弃 |
| `@list-empty-text-padding` | - | 由于样式变化已废弃 |
| `@list-item-padding` | `listItemPadding` | - |
| `@list-item-padding-sm` | `listItemPaddingSM` | - |
| `@list-item-padding-lg` | `listItemPaddingLG` | - |
| `@list-item-meta-margin-bottom` | - | 由于样式变化已废弃 |
| `@list-item-meta-avatar-margin-right` | - | 由于样式变化已废弃 |
| `@list-item-meta-title-margin-bottom` | - | 由于样式变化已废弃 |
| `@list-customize-card-bg` | - | 由于样式变化已废弃 |
| `@list-item-meta-description-font-size` | - | 由于样式变化已废弃 |
```
List 在 v5 没有样式变化，这些 token 应该在迁移过程中丢掉了。
可以对照 v4 里 less 变量的用法修改一下样式，把这些 token 加回来，一般来说只是被替换成了某个 v5 的 token，改成用这个 token 初始化 component token 即可

Comment (jrr997) [reply docs/react/migrate-less-variables.zh-CN.md:25]: OK,我改一下。

**Comment by jrr997** [comment]
迁移后发现一个问题，List 的 `bordered === true` 时，`listItemPaddingLG` 不生效（`listItemPaddingSM`是生效的）。看了代码，有 border 时 List 的 padding 没有取 `listItemPaddingLG`，感觉是 v4 迁移到 v5 时迁错了？
相关代码在：  https://github.com/ant-design/ant-design/blob/7c628b0a57d47e23f5c6989f4fa7ab6d9cb5e2f8/components/list/style/index.ts#L56

File: docs/react/migrate-less-variables.zh-CN.md [hunk 2]
```diff
@@ -64,7 +64,22 @@ title: Less 变量迁移 Design Token
 
 <!-- ### Layout 布局 -->
 
-<!-- ### List 列表 -->
+### List 列表
+
+<!-- prettier-ignore -->
+| less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@list-header-background` | `headerBackground` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:72]:
```suggestion
| `@list-header-background` | `headerBg` | - |
```

File: docs/react/migrate-less-variables.zh-CN.md [hunk 3]
```diff
@@ -64,7 +64,22 @@ title: Less 变量迁移 Design Token
...
+| `@list-footer-background` | `footerBackground` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:73]:
```suggestion
| `@list-footer-background` | `footerBg` | - |
```

**Comment by MadCcc** [comment]
> listItemPaddingLG

迁移之后丢掉了也有可能，对照 v4 如果是用了 token 的话可以改回去的

File: components/list/style/index.ts [hunk 4]
```diff
@@ -5,14 +5,21 @@ import { genComponentStyleHook, mergeToken } from '../../theme/internal';
 
 export interface ComponentToken {
   contentWidth: number;
+  itemPaddingLG: string;
+  itemPaddingSM: string;
+  itemPadding: string;
+  headerBackground: string;
+  footerBackground: string;
+  emptyTextPadding: number;
```
Comment (MadCcc) [inline components/list/style/index.ts:13]:
```suggestion
  emptyTextPadding: CSSProperties['padding'];
```
支持一下字符串

Comment (MadCcc) [reply components/list/style/index.ts:13]: 其他不用于计算的间距类 token 都可以改一下

**Review by MadCcc — APPROVED** [review]

**Pull request merged**